qa/tasks/deepsea.py

```diff
@@ -877,62 +881,59 @@ def _run_stage_5(self):
         self._run_orch(("stage", 5))
 
     def begin(self):
-        self.logger.debug("beginning of begin method")
+        self.log.debug("beginning of begin method")
         if self.state_orch:
-            self.logger.info(
-                "WWWW: running orchestration {}".format(self.state_orch)
-                )
+            self.log.info(anchored(
+                "running orchestration {}".format(self.state_orch)
+                ))
             self._run_orch(("orch", self.state_orch))
-            self.logger.debug("end of begin method")
+            self.log.debug("end of begin method")
             return None
         # it's not an orch, so it must be a stage
         assert self.stage, "Neither state_orch, nor stage"
         if self._is_int_between_0_and_5(self.stage):
             exec('self._run_stage_{}()'.format(self.stage))
         elif self.stage == 'prep':
-            self.logger.info("Running Stage 0 instead of Stage \"prep\"")
+            self.log.info("Running Stage 0 instead of Stage \"prep\"")
             self._run_stage_0()
         elif self.stage == 'discovery':
-            self.logger.info("Running Stage 1 instead of Stage \"discovery\"")
+            self.log.info("Running Stage 1 instead of Stage \"discovery\"")
             self._run_stage_1()
         elif self.stage == 'configure':
-            self.logger.info("Running Stage 2 instead of Stage \"configure\"")
+            self.log.info("Running Stage 2 instead of Stage \"configure\"")
             self._run_stage_2()
         elif self.stage == 'deploy':
-            self.logger.info("Running Stage 3 instead of Stage \"deploy\"")
+            self.log.info("Running Stage 3 instead of Stage \"deploy\"")
             self._run_stage_3()
         elif self.stage == 'services':
-            self.logger.info("Running Stage 4 instead of Stage \"services\"")
+            self.log.info("Running Stage 4 instead of Stage \"services\"")
             self._run_stage_4()
         elif self.stage == 'removal':
-            self.logger.info("Running Stage 5 instead of Stage \"removal\"")
+            self.log.info("Running Stage 5 instead of Stage \"removal\"")
             self._run_stage_5()
         elif self.stage in self.all_stages:
-            self.logger.info(
+            self.log.info(
                 "Running non-numeric Stage \"{}\"".format(self.stage)
```

Comment:
I would suggest to unify logging and code duplication using the code:
```python
    synonyms = {'prep': '0', 'discovery': '1', 'configure': '2', 'deploy': '3', 'services': '4', 'removal': '5'}

    if self.stage not in self.all_stages:
        raise self.ConfigError('Unknown stage "%s"' % self.stage)

    stage = synonyms.get(self.stage, self.stage)
    self.log.info('Running Stage %s' % stage)

    # next if is not required if to have _run_stage_<stage_name> for each stage
    if stage not in [str(x) for x in range(0, 6)]:
        self._run_orch(("stage", stage))
        return
    mname = "_run_stage_%s" % stage
    if not hasattr(self, mname):
       raise Exception("Can't find method '%s'" % mname)
    run_stage = getattr(self, mname)
    if not callable(run_stage):
       raise Exception("The '%s' is not callable method" % mname)
    run_stage()    
```

Reply:
If information about synonyms is important then instead of:
```
    self.log.info('Running Stage %s' % stage)
```
following can be written:
```
if self.stage in synonyms:
    self.log.info('Running Stage %s instead of Stage "%s"' % (stage, self.stage))
else:
    self.log.info('Running Stage %s' % stage)
```

Reply:
I prefer the new `.format` over the old `%`

Reply:
I don't agree that there is code duplication. DeepSea has Stages 0 through 5, period. It does not have a variable number of stages. The fact that the code handles each stage explicitly can be seen as an advantage.

Reply:
> I prefer the new `.format` over the old `%`
1) % is old but not bad, and still valid in all python versions.
2) the idea of my comments is not in making you to use '%' over '.format', I just use it to type less characters, and code looks more compact and clear.
3) there is no sense in expected from my comments that you can just copy and paste them into your code.

Reply:
> I don't agree that there is code duplication. DeepSea has Stages 0 through 5, period. It does not have a variable number of stages. The fact that the code handles each stage explicitly can be seen as an advantage.

Come on, you have already variable number of stage, like numbered and non-numbered, you've listed them in **all_stages** list. The idea of **UNIFYING** is to have the same reporting format, to ease log parsing for detect substages. Otherwise I don't understand why you have 6-7 same logging info? What is this if not duplication? More over, if you'd care about not breaking the current CI parsers, you'd left "Running DeepSea Stage X" format. But now we need to invent something convenient in order to better detect point of failure. However, I guess we can use, natural teuthology "Running task" markers, which is even better.

All that said is not something as mandatory changes required or 'really must be', it's just suggestion.

Reply:
Whoa! In my own defense, this is the first time you've told me what string the CI parser looks for. I'll push a patch to address this regrettable "not caring" of mine.

Reply:
Please note that this task is under development and the code is in flux. Your input is appreciated, but do please be patient with me as I endeavor to incorporate it as best I can.

Reply:
> Please note that this task is under development and the code is in flux. Your input is appreciated, but do please be patient with me as I endeavor to incorporate it as best I can.

Sure, we are developers, we develop code :-)

Reply:
> Whoa! In my own defense, this is the first time you've told me what string the CI parser looks for. I'll push a patch to address this regrettable "not caring" of mine.

Not the first time, second one, earlier I've sent a link to you to the parser code: https://github.com/SUSE/sesci/blob/master/jenkins/util, it's there in  `make_brief_deepsea_report` ;-)

Reply:
Keep calm and _do the code_.

Reply:
@kshtsk 

!(Dont Panic)[https://tinyurl.com/yahplycd]